Creating Stories for Card, Banner, Media with Caption.

jenbreese opened the pull request:
# READY FOR REVIEW

# Summary
- Added a Banner story
- Added a Media with Caption story
- Added classes to the banner 
- Updated the button classes
- Changed the color from dark red to the bright red in the main nav and the brandbar
- Updated the lockup classes with size and spacing to Line 5. Added the border top when mobile to Line 5.

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to the storybook and look at the elements in the summary above.
3. Verify it work as expected.

jenbreese committed:
Banner changes

jenbreese committed:
fixup to the media with caption

jenbreese committed:
adding the composition to comply with lint

jenbreese committed:
removed experiment for lint

jenbreese committed:
added the composition

jenbreese committed:
merging

pookmish commented:
I'm not sure how much details the VS code in gitpod provides. My phpstorm tells me this:
`is missing the following properties from type ParagraphInterface: created, langcode`

You can look at the card.stories.tsx and see how the `composition`, `langcode`, and `created` data looks. You can even copy and paste it into your story.

jenbreese committed:
fixup for the error

jenbreese committed:
Merge branch 'main' into FE-learning

jenbreese committed:
padding and font size changes

jenbreese committed:
added typename and classes

jenbreese committed:
Merge branch 'main' into FE-learning

jenbreese commented:
I added classes for the Media caption. I added the `__typename` to the story. I haven't figured out why the caption and link aren't appearing on the nextjs version of the site.

pookmish commented:
@jenbreese I don't understand what you mean by "nextjs version of the site". It looks like your story book story is working correctly for the media with caption. 
![image](https://github.com/pookmish/cardinal-sites-next/assets/7185045/bcb34acb-0d60-42e4-b4fc-6f6d5fd6baca)

jenbreese commented:
I thought it has to work on the Storybook (port 6006) and the port 3000 site where the file `src/components/paragraphs/stanford-media-caption/media-caption-paragraph.tsx` gets used? I turned off my GitPod so I don't have a way to screenshot it now.

pookmish commented:
Oh thank you. It was missing from the GQL queries. You can see where I added the missing fields in the fragments here: https://github.com/pookmish/cardinal-sites-next/commit/d5e529744f8ab8682d8a1dfef0f290be2f6a84c5

jenbreese committed:
Merge branch 'main' into FE-learning

jenbreese commented:
@pookmish Thank you thank you thank you.

jenbreese committed:
Adding changes to the Lockup in the header

jenbreese committed:
merge

jenbreese committed:
change to modular type size

jenbreese commented on src/components/paragraphs/stanford-banner/banner-paragraph.tsx:
```diff
@@ -60,12 +60,12 @@ const BannerParagraph = ({paragraph, eagerLoadImage, ...props}: Props) => {
             }
 
-            <Wysiwyg html={paragraph.suBannerBody?.processed} className="order-3"/>
+            <Wysiwyg html={paragraph.suBannerBody?.processed} className="order-3 text-m0"/>
```
There is a 100ch width applied to the p within this. It keeps it from aligning flush right to the image edge. @pookmish Do you know the best way to fix that?

pookmish replied:
I'll fix in another commit

pookmish replied:
I'm not sure I know what you mean. when should the text align to the image edge of the banner?

jenbreese replied:
I'm sorry @pookmish That comment should have gone on the media with caption file.

https://github.com/pookmish/cardinal-sites-next/blob/a440083e5492da15e08caa664c6c2f3ec093bc46/src/components/paragraphs/stanford-media-caption/media-caption-paragraph.tsx#L40

pookmish replied:
Ok, but from what I see, it is aligning to the right of the image. I'm not sure what you are seeing different 
![image](https://github.com/pookmish/cardinal-sites-next/assets/7185045/c50f7a3e-9bf8-4f19-86ab-ac7840ba9e08)

jenbreese replied:
Well, it is right aligned fine now for me too.  🤷🏼‍♀️
